Automattic/simplenote-macos
# Implement API-based triggered CI deployment for the beta build

This is based off #927 for ease of review. That PR needs to be merged and this one rebased before merging.

The branch name is not the best... I meant to move the final work into a new one, but I realized I didn't only after creating this PR.

---

### Fix

Implements the CircleCI API based trigger for the build and deployment of the beta app to App Center.

### Test

1. Checkout this branch
2. Run `bundle install`
3. Run `bundle exec fastlane trigger_beta_ci_build branch_to_build:fix-internal-app-automation-wip`

You should see a new build starting [in CI](https://app.circleci.com/pipelines/github/Automattic/simplenote-macos)

![image](https://user-images.githubusercontent.com/1218433/115824229-fa5f0b00-a44a-11eb-9803-3c4089581f37.png)

The build should pass and you should then see a new build available for distribution on App Center. You may want to download it and launch it, to ensure the notarization worked. (_Please deleted it from App Center once verified it, just so the list is clean_)

![image](https://user-images.githubusercontent.com/1218433/115826024-a275d380-a44d-11eb-861d-b8141d5cbdb8.png)

**Extra verification** Please try to run the build locally. While working on this I experienced code sign issues from Xcode, but I'm not sure if it was just Xcode not being in sync with the keychain or with the Developer Portal, or if there was actually something broken. The app builds and run on my Mac right now.

### Review
Only one developer required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.
